src/components/TodoItem.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import styled, { keyframes } from "styled-components";
+
+const TodoItem = ({ id, isDone, content, date, onUpdate, removeTodo }) => {
+  const [isRemoving, setIsRemoving] = useState(false); // 삭제 애니메이션 상태
+
+  const onChangeCheckbox = () => {
+    onUpdate(id);
+  };
+
+  const handleRemove = () => {
+    setIsRemoving(true); // 삭제 애니메이션 시작
+    setTimeout(() => {
+      removeTodo(id); // 애니메이션 후 삭제
+    }, 200); // 애니메이션이 끝난 후 200ms 대기 후 삭제
+  };
+
+  return (
+    <TodoItemContainer isRemoving={isRemoving}>
+      <Checkbox
+        type="checkbox"
+        id={id}
+        checked={isDone}
+        onChange={onChangeCheckbox}
+      />
+      <Content className={`content ${isDone ? "completed" : ""}`}>
+        {content}
+      </Content>
+      <DateText>{new Date(date).toLocaleDateString()}</DateText>
+      <DeleteButton onClick={handleRemove}>Del</DeleteButton>
+    </TodoItemContainer>
+  );
+};
+
+// 할 일 추가 애니메이션 정의
+const slideDownFadeIn = keyframes`
+  from {
+    opacity: 0;
+    transform: translateY(-20px);
+  }
+  to {
+    opacity: 1;
+    transform: translateY(0);
+  }
+`;
+
+// 할 일 삭제 애니메이션 정의
+const fadeOutScaleDown = keyframes`
+  from {
+    opacity: 1;
+    transform: scale(1);
+  }
+  to {
+    opacity: 0;
+    transform: scale(0.9) translateY(+10px);
+  }
+`;
+
+const TodoItemContainer = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 20px;
+  padding-bottom: 20px;
+  border-bottom: 1px solid rgb(240, 240, 240);
+  animation: ${({ isRemoving }) =>
+      isRemoving ? fadeOutScaleDown : slideDownFadeIn}
+    0.3s ease-out forwards;
```

Comment:
애니메이션에서 시간을 정의하는 부분이 하드코딩으로 들어가있는데 이번 과제처럼 간단한 애니메이션 구현에서는 전혀 문제가 되지 않지만 같은 수 또는 문자열 등을 여러 번 사용해야하는 프로젝트에서는 상수를 따로 정의해주는 게 가독성이나 유지보수 측면에서 훨씬 나을거예요! 위에서 사용하신 handleRemove 함수에서도 마찬가지로 상수를 사용하면 `// 애니메이션이 끝난 후 200ms 대기 후 삭제` 라는 주석도 쓰지 않아도 되겠죠!

```suggestion
const ANIMATION_TIME = 300;

animation: ${({ isRemoving }) =>
      isRemoving ? fadeOutScaleDown : slideDownFadeIn}
    ${ANIMATION_TIME}ms ease-out forwards;
```